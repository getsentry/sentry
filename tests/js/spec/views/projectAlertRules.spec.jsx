import React from 'react';
import {mount} from 'enzyme';

import ProjectAlertRules from 'app/views/projectAlertRules';

describe('projectAlertRules', function() {
  beforeEach(function() {
    MockApiClient.addMockResponse({
      url: '/projects/org1/project1/rules/',
      body: [TestStubs.ProjectAlertRule()],
    });
  });

  afterEach(function() {
    MockApiClient.clearMockResponses();
  });

  it('renders', function() {
    const routes = [];

    const wrapper = mount(
      <ProjectAlertRules
        routes={routes}
        params={{orgId: 'org1', projectId: 'project1'}}
      />,
      TestStubs.routerContext()
    );
    expect(wrapper).toMatchSnapshot();
  });
});
