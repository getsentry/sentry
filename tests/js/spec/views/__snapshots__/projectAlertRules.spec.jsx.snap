// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projectAlertRules renders 1`] = `
<ProjectAlertRules
  params={
    Object {
      "orgId": "org1",
      "projectId": "project1",
    }
  }
  routes={Array []}
>
  <SideEffect(DocumentTitle)
    title="Sentry"
  >
    <DocumentTitle
      title="Sentry"
    >
      <SettingsPageHeading
        action={
          <Tooltip
            disabled={true}
            title="You do not have permission to edit alert rules."
          >
            <Button
              disabled={false}
              icon="icon-circle-add"
              priority="primary"
              size="small"
              to="new/"
            >
              New Alert Rule
            </Button>
          </Tooltip>
        }
        tabs={
          <ul
            className="nav nav-tabs"
            style={
              Object {
                "borderBottom": "1px solid #ddd",
              }
            }
          >
            <ListLink
              activeClassName="active"
              index={false}
              to="alerts"
            >
              Settings
            </ListLink>
            <ListLink
              activeClassName="active"
              index={false}
              to=""
            >
              Rules
            </ListLink>
          </ul>
        }
        title="Alerts"
      >
        <Wrapper
          tabs={
            <ul
              className="nav nav-tabs"
              style={
                Object {
                  "borderBottom": "1px solid #ddd",
                }
              }
            >
              <ListLink
                activeClassName="active"
                index={false}
                to="alerts"
              >
                Settings
              </ListLink>
              <ListLink
                activeClassName="active"
                index={false}
                to=""
              >
                Rules
              </ListLink>
            </ul>
          }
        >
          <div
            className="css-1pnrn3o-Wrapper e1kblvez1"
          >
            <Flex
              align="center"
            >
              <Base
                align="center"
                className="css-5ipae5"
              >
                <div
                  className="css-5ipae5"
                  is={null}
                >
                  <Title>
                    <Base
                      className="css-1mqcea-Title e1kblvez0"
                    >
                      <div
                        className="css-1mqcea-Title e1kblvez0"
                        is={null}
                      >
                        Alerts
                      </div>
                    </Base>
                  </Title>
                  <div>
                    <Tooltip
                      disabled={true}
                      title="You do not have permission to edit alert rules."
                    >
                      <Button
                        disabled={false}
                        icon="icon-circle-add"
                        priority="primary"
                        size="small"
                        to="new/"
                      >
                        <StyledButton
                          aria-label="New Alert Rule"
                          disabled={false}
                          onClick={[Function]}
                          priority="primary"
                          role="button"
                          size="small"
                          to="new/"
                        >
                          <Component
                            aria-label="New Alert Rule"
                            className="css-zvpqlo-StyledButton-getColors e17811v30"
                            disabled={false}
                            onClick={[Function]}
                            priority="primary"
                            role="button"
                            size="small"
                            to="new/"
                          >
                            <Link
                              aria-label="New Alert Rule"
                              className="css-zvpqlo-StyledButton-getColors e17811v30"
                              disabled={false}
                              onClick={[Function]}
                              onlyActiveOnIndex={false}
                              priority="primary"
                              role="button"
                              size="small"
                              style={Object {}}
                              to="new/"
                            >
                              <a
                                aria-label="New Alert Rule"
                                className="css-zvpqlo-StyledButton-getColors e17811v30"
                                disabled={false}
                                onClick={[Function]}
                                priority="primary"
                                role="button"
                                size="small"
                                style={Object {}}
                              >
                                <ButtonLabel
                                  size="small"
                                >
                                  <Component
                                    className="css-19bmqb3-ButtonLabel e17811v31"
                                    size="small"
                                  >
                                    <Flex
                                      align="center"
                                      className="css-19bmqb3-ButtonLabel e17811v31"
                                      size="small"
                                    >
                                      <Base
                                        align="center"
                                        className="e17811v31 css-1eblpyp-ButtonLabel"
                                        size="small"
                                      >
                                        <div
                                          className="e17811v31 css-1eblpyp-ButtonLabel"
                                          is={null}
                                          size="small"
                                        >
                                          <Icon
                                            hasChildren={true}
                                            size="small"
                                          >
                                            <Component
                                              className="css-1vdnsie-Icon e17811v32"
                                              hasChildren={true}
                                              size="small"
                                            >
                                              <Box
                                                className="css-1vdnsie-Icon e17811v32"
                                                size="small"
                                              >
                                                <Base
                                                  className="e17811v32 css-nk081r-Icon"
                                                  size="small"
                                                >
                                                  <div
                                                    className="e17811v32 css-nk081r-Icon"
                                                    is={null}
                                                    size="small"
                                                  >
                                                    <StyledInlineSvg
                                                      size="12px"
                                                      src="icon-circle-add"
                                                    >
                                                      <InlineSvg
                                                        className="css-1ov3rcq-StyledInlineSvg e17811v33"
                                                        size="12px"
                                                        src="icon-circle-add"
                                                      >
                                                        <StyledSvg
                                                          className="css-1ov3rcq-StyledInlineSvg e17811v33"
                                                          height="12px"
                                                          viewBox={Object {}}
                                                          width="12px"
                                                        >
                                                          <svg
                                                            className="e17811v33 css-1jjmnki-StyledSvg-StyledInlineSvg e2idor0"
                                                            height="12px"
                                                            viewBox={Object {}}
                                                            width="12px"
                                                          >
                                                            <use
                                                              href="#test"
                                                              xlinkHref="#test"
                                                            />
                                                          </svg>
                                                        </StyledSvg>
                                                      </InlineSvg>
                                                    </StyledInlineSvg>
                                                  </div>
                                                </Base>
                                              </Box>
                                            </Component>
                                          </Icon>
                                          New Alert Rule
                                        </div>
                                      </Base>
                                    </Flex>
                                  </Component>
                                </ButtonLabel>
                              </a>
                            </Link>
                          </Component>
                        </StyledButton>
                      </Button>
                    </Tooltip>
                  </div>
                </div>
              </Base>
            </Flex>
            <div>
              <ul
                className="nav nav-tabs"
                style={
                  Object {
                    "borderBottom": "1px solid #ddd",
                  }
                }
              >
                <ListLink
                  activeClassName="active"
                  index={false}
                  to="alerts"
                >
                  <li
                    className=""
                  >
                    <Link
                      onlyActiveOnIndex={false}
                      style={Object {}}
                      to="alerts"
                    >
                      <a
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Settings
                      </a>
                    </Link>
                  </li>
                </ListLink>
                <ListLink
                  activeClassName="active"
                  index={false}
                  to=""
                >
                  <li
                    className=""
                  >
                    <Link
                      onlyActiveOnIndex={false}
                      style={Object {}}
                      to=""
                    >
                      <a
                        style={Object {}}
                      >
                        Rules
                      </a>
                    </Link>
                  </li>
                </ListLink>
              </ul>
            </div>
          </div>
        </Wrapper>
      </SettingsPageHeading>
      <div
        className="rules-list"
      >
        <RuleRow
          canEdit={true}
          data={
            Object {
              "actions": Array [
                Object {
                  "id": "sentry.rules.actions.notify1",
                  "name": "Send a notification to all services",
                },
              ],
              "conditions": Array [
                Object {
                  "id": "sentry.rules.conditions.1",
                  "name": "An alert is first seen",
                },
              ],
              "environment": "staging",
              "id": "1",
              "name": "My alert rule",
            }
          }
          firstRule={true}
          key="1"
          onDelete={[Function]}
          orgId="org1"
          params={
            Object {
              "orgId": "org1",
              "projectId": "project1",
            }
          }
          projectId="project1"
          routes={Array []}
        >
          <Panel>
            <Component
              className="css-10qfvek-Panel e1laxa7d0"
            >
              <div
                className="css-10qfvek-Panel e1laxa7d0"
              >
                <PanelHeader
                  align="center"
                  className="css-1lx593h"
                  justify="space-between"
                >
                  <Component
                    align="center"
                    className="css-10edylt-PanelHeader-getPadding e1p8v8nv0"
                    justify="space-between"
                  >
                    <Flex
                      align="center"
                      className="css-10edylt-PanelHeader-getPadding e1p8v8nv0"
                      justify="space-between"
                    >
                      <Base
                        align="center"
                        className="e1p8v8nv0 css-51qtoa-PanelHeader-getPadding"
                        justify="space-between"
                      >
                        <div
                          className="e1p8v8nv0 css-51qtoa-PanelHeader-getPadding"
                          is={null}
                        >
                          <TextColorLink
                            to="1/"
                          >
                            <Link
                              className="css-1s5bpjc-TextColorLink e1abl5wd0"
                              onlyActiveOnIndex={false}
                              style={Object {}}
                              to="1/"
                            >
                              <a
                                className="css-1s5bpjc-TextColorLink e1abl5wd0"
                                onClick={[Function]}
                                style={Object {}}
                              >
                                My alert rule
                                 - 
                                All Environments
                              </a>
                            </Link>
                          </TextColorLink>
                          <div>
                            <Tooltip
                              disabled={true}
                              title="You do not have permission to edit alert rules."
                            >
                              <Button
                                data-test-id="edit-rule"
                                disabled={false}
                                size="small"
                                style={
                                  Object {
                                    "marginRight": 5,
                                  }
                                }
                                to="1/"
                              >
                                <StyledButton
                                  aria-label="Edit Rule"
                                  data-test-id="edit-rule"
                                  disabled={false}
                                  onClick={[Function]}
                                  role="button"
                                  size="small"
                                  style={
                                    Object {
                                      "marginRight": 5,
                                    }
                                  }
                                  to="1/"
                                >
                                  <Component
                                    aria-label="Edit Rule"
                                    className="css-dkprmi-StyledButton-getColors e17811v30"
                                    data-test-id="edit-rule"
                                    disabled={false}
                                    onClick={[Function]}
                                    role="button"
                                    size="small"
                                    style={
                                      Object {
                                        "marginRight": 5,
                                      }
                                    }
                                    to="1/"
                                  >
                                    <Link
                                      aria-label="Edit Rule"
                                      className="css-dkprmi-StyledButton-getColors e17811v30"
                                      data-test-id="edit-rule"
                                      disabled={false}
                                      onClick={[Function]}
                                      onlyActiveOnIndex={false}
                                      role="button"
                                      size="small"
                                      style={
                                        Object {
                                          "marginRight": 5,
                                        }
                                      }
                                      to="1/"
                                    >
                                      <a
                                        aria-label="Edit Rule"
                                        className="css-dkprmi-StyledButton-getColors e17811v30"
                                        data-test-id="edit-rule"
                                        disabled={false}
                                        onClick={[Function]}
                                        role="button"
                                        size="small"
                                        style={
                                          Object {
                                            "marginRight": 5,
                                          }
                                        }
                                      >
                                        <ButtonLabel
                                          size="small"
                                        >
                                          <Component
                                            className="css-19bmqb3-ButtonLabel e17811v31"
                                            size="small"
                                          >
                                            <Flex
                                              align="center"
                                              className="css-19bmqb3-ButtonLabel e17811v31"
                                              size="small"
                                            >
                                              <Base
                                                align="center"
                                                className="e17811v31 css-1eblpyp-ButtonLabel"
                                                size="small"
                                              >
                                                <div
                                                  className="e17811v31 css-1eblpyp-ButtonLabel"
                                                  is={null}
                                                  size="small"
                                                >
                                                  Edit Rule
                                                </div>
                                              </Base>
                                            </Flex>
                                          </Component>
                                        </ButtonLabel>
                                      </a>
                                    </Link>
                                  </Component>
                                </StyledButton>
                              </Button>
                            </Tooltip>
                            <Tooltip
                              disabled={true}
                              title="You do not have permission to edit alert rules."
                            >
                              <Confirm
                                cancelText="Cancel"
                                confirmText="Confirm"
                                disableConfirmButton={false}
                                disabled={false}
                                message="Are you sure you want to remove this rule?"
                                onConfirm={[Function]}
                                priority="primary"
                              >
                                <Button
                                  disabled={false}
                                  icon="icon-trash"
                                  onClick={[Function]}
                                  size="small"
                                >
                                  <StyledButton
                                    disabled={false}
                                    onClick={[Function]}
                                    role="button"
                                    size="small"
                                  >
                                    <Component
                                      className="css-dkprmi-StyledButton-getColors e17811v30"
                                      disabled={false}
                                      onClick={[Function]}
                                      role="button"
                                      size="small"
                                    >
                                      <button
                                        className="css-dkprmi-StyledButton-getColors e17811v30"
                                        disabled={false}
                                        onClick={[Function]}
                                        role="button"
                                        size="small"
                                      >
                                        <ButtonLabel
                                          size="small"
                                        >
                                          <Component
                                            className="css-19bmqb3-ButtonLabel e17811v31"
                                            size="small"
                                          >
                                            <Flex
                                              align="center"
                                              className="css-19bmqb3-ButtonLabel e17811v31"
                                              size="small"
                                            >
                                              <Base
                                                align="center"
                                                className="e17811v31 css-1eblpyp-ButtonLabel"
                                                size="small"
                                              >
                                                <div
                                                  className="e17811v31 css-1eblpyp-ButtonLabel"
                                                  is={null}
                                                  size="small"
                                                >
                                                  <Icon
                                                    hasChildren={false}
                                                    size="small"
                                                  >
                                                    <Component
                                                      className="css-ljhpxy-Icon e17811v32"
                                                      hasChildren={false}
                                                      size="small"
                                                    >
                                                      <Box
                                                        className="css-ljhpxy-Icon e17811v32"
                                                        size="small"
                                                      >
                                                        <Base
                                                          className="e17811v32 css-7mohvl-Icon"
                                                          size="small"
                                                        >
                                                          <div
                                                            className="e17811v32 css-7mohvl-Icon"
                                                            is={null}
                                                            size="small"
                                                          >
                                                            <StyledInlineSvg
                                                              size="12px"
                                                              src="icon-trash"
                                                            >
                                                              <InlineSvg
                                                                className="css-1ov3rcq-StyledInlineSvg e17811v33"
                                                                size="12px"
                                                                src="icon-trash"
                                                              >
                                                                <StyledSvg
                                                                  className="css-1ov3rcq-StyledInlineSvg e17811v33"
                                                                  height="12px"
                                                                  viewBox={Object {}}
                                                                  width="12px"
                                                                >
                                                                  <svg
                                                                    className="e17811v33 css-1jjmnki-StyledSvg-StyledInlineSvg e2idor0"
                                                                    height="12px"
                                                                    viewBox={Object {}}
                                                                    width="12px"
                                                                  >
                                                                    <use
                                                                      href="#test"
                                                                      xlinkHref="#test"
                                                                    />
                                                                  </svg>
                                                                </StyledSvg>
                                                              </InlineSvg>
                                                            </StyledInlineSvg>
                                                          </div>
                                                        </Base>
                                                      </Box>
                                                    </Component>
                                                  </Icon>
                                                </div>
                                              </Base>
                                            </Flex>
                                          </Component>
                                        </ButtonLabel>
                                      </button>
                                    </Component>
                                  </StyledButton>
                                </Button>
                                <Modal
                                  animation={false}
                                  autoFocus={true}
                                  backdrop={true}
                                  bsClass="modal"
                                  dialogComponentClass={[Function]}
                                  enforceFocus={true}
                                  keyboard={true}
                                  manager={
                                    ModalManager {
                                      "add": [Function],
                                      "containers": Array [],
                                      "data": Array [],
                                      "handleContainerOverflow": true,
                                      "hideSiblingNodes": true,
                                      "isTopModal": [Function],
                                      "modals": Array [],
                                      "remove": [Function],
                                    }
                                  }
                                  onHide={[Function]}
                                  renderBackdrop={[Function]}
                                  restoreFocus={true}
                                  show={false}
                                >
                                  <Modal
                                    autoFocus={true}
                                    backdrop={true}
                                    backdropClassName="modal-backdrop"
                                    containerClassName="modal-open"
                                    enforceFocus={true}
                                    keyboard={true}
                                    manager={
                                      ModalManager {
                                        "add": [Function],
                                        "containers": Array [],
                                        "data": Array [],
                                        "handleContainerOverflow": true,
                                        "hideSiblingNodes": true,
                                        "isTopModal": [Function],
                                        "modals": Array [],
                                        "remove": [Function],
                                      }
                                    }
                                    onEntering={[Function]}
                                    onExited={[Function]}
                                    onHide={[Function]}
                                    renderBackdrop={[Function]}
                                    restoreFocus={true}
                                    show={false}
                                  />
                                </Modal>
                              </Confirm>
                            </Tooltip>
                          </div>
                        </div>
                      </Base>
                    </Flex>
                  </Component>
                </PanelHeader>
                <PanelBody
                  direction="column"
                  disablePadding={true}
                  flex={false}
                >
                  <div
                    className="css-9vq8an-textStyles"
                  >
                    <RuleDescriptionRow>
                      <div
                        className="css-16pvk6d-RuleDescriptionRow e1abl5wd1"
                      >
                        <RuleDescriptionColumn>
                          <div
                            className="css-15obcn9-RuleDescriptionColumn e1abl5wd2"
                          >
                            <Condition>
                              <div
                                className="css-1d9hhun-Condition e1abl5wd3"
                              >
                                <h6>
                                  When 
                                  <strong />
                                   of these conditions are met:
                                </h6>
                                <GuideAnchor
                                  target="alert_conditions"
                                  type="text"
                                >
                                  <GuideAnchorContainer
                                    innerRef={[Function]}
                                    type="text"
                                  >
                                    <div
                                      className="css-9u5for-GuideAnchorContainer e130o4350"
                                      type="text"
                                    >
                                      <table
                                        className="conditions-list table"
                                      >
                                        <tbody>
                                          <tr
                                            key="0"
                                          >
                                            <td>
                                              An alert is first seen
                                            </td>
                                          </tr>
                                        </tbody>
                                      </table>
                                      <StyledGuideAnchor
                                        active={false}
                                        className="guide-anchor-ping alert_conditions"
                                      >
                                        <div
                                          className="guide-anchor-ping alert_conditions css-1yndvnf-StyledGuideAnchor e130o4351"
                                        >
                                          <StyledGuideAnchorRipples>
                                            <div
                                              className="css-3zj3g7-StyledGuideAnchorRipples e130o4352"
                                            />
                                          </StyledGuideAnchorRipples>
                                        </div>
                                      </StyledGuideAnchor>
                                    </div>
                                  </GuideAnchorContainer>
                                </GuideAnchor>
                              </div>
                            </Condition>
                          </div>
                        </RuleDescriptionColumn>
                        <RuleDescriptionColumn>
                          <div
                            className="css-15obcn9-RuleDescriptionColumn e1abl5wd2"
                          >
                            <Condition>
                              <div
                                className="css-1d9hhun-Condition e1abl5wd3"
                              >
                                <h6>
                                  Take these actions at most
                                   
                                  <strong>
                                    once every 
                                    <Duration
                                      seconds={NaN}
                                    >
                                      <span>
                                        NaN ms
                                      </span>
                                    </Duration>
                                  </strong>
                                   
                                  for an issue:
                                </h6>
                                <GuideAnchor
                                  target="alert_actions"
                                  type="text"
                                >
                                  <GuideAnchorContainer
                                    innerRef={[Function]}
                                    type="text"
                                  >
                                    <div
                                      className="css-9u5for-GuideAnchorContainer e130o4350"
                                      type="text"
                                    >
                                      <table
                                        className="actions-list table"
                                      >
                                        <tbody>
                                          <tr
                                            key="0"
                                          >
                                            <td>
                                              Send a notification to all services
                                            </td>
                                          </tr>
                                        </tbody>
                                      </table>
                                      <StyledGuideAnchor
                                        active={false}
                                        className="guide-anchor-ping alert_actions"
                                      >
                                        <div
                                          className="guide-anchor-ping alert_actions css-1yndvnf-StyledGuideAnchor e130o4351"
                                        >
                                          <StyledGuideAnchorRipples>
                                            <div
                                              className="css-3zj3g7-StyledGuideAnchorRipples e130o4352"
                                            />
                                          </StyledGuideAnchorRipples>
                                        </div>
                                      </StyledGuideAnchor>
                                    </div>
                                  </GuideAnchorContainer>
                                </GuideAnchor>
                              </div>
                            </Condition>
                          </div>
                        </RuleDescriptionColumn>
                      </div>
                    </RuleDescriptionRow>
                  </div>
                </PanelBody>
              </div>
            </Component>
          </Panel>
        </RuleRow>
      </div>
    </DocumentTitle>
  </SideEffect(DocumentTitle)>
</ProjectAlertRules>
`;
